mobile/lib/features/wallet/domain/wallet_history.dart

```diff
@@ -0,0 +1,77 @@
+import 'package:get_10101/common/domain/model.dart';
+import 'payment_flow.dart';
+import 'package:get_10101/bridge_generated/bridge_definitions.dart' as rust;
+
+enum WalletHistoryItemDataType { lightning, onChain, trade }
+
+enum WalletHistoryStatus { pending, confirmed }
+
+class WalletHistoryItemData {
+  final PaymentFlow flow;
+  final Amount amount;
+  final WalletHistoryItemDataType type;
+  final WalletHistoryStatus status;
+  final DateTime timestamp;
+
+  // on-chain
+  final String? txid;
+
+  // lightning
+  final String? paymentHash;
+
+  // trade
+  final String? orderId;
+
```

Comment:
this shows rather well that these should rather be distinct types on the domain level 🙃